source/core/src/main/com/deco2800/game/components/player/PlayerMeleeAttackComponent.java

```diff
@@ -0,0 +1,275 @@
+import com.deco2800.game.components.PlayerCombatStatsComponent;
+import com.deco2800.game.entities.Entity;
+import com.deco2800.game.physics.BodyUserData;
+import com.deco2800.game.physics.PhysicsLayer;
+import com.deco2800.game.physics.components.PhysicsComponent;
+import com.deco2800.game.physics.components.PhysicsComponent.AlignX;
+import com.deco2800.game.physics.components.PhysicsComponent.AlignY;
+import com.deco2800.game.rendering.AnimationRenderComponent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PlayerMeleeAttackComponent extends Component {
+    private static final Logger logger = LoggerFactory.getLogger(PlayerMeleeAttackComponent.class);
+
+    private final FixtureDef fixtureDef;
+    private Fixture fixture;
+    private PlayerCombatStatsComponent playerCombatStats;
+    private PlayerMeleeAttackComponent playerAttackComponent;
+    private boolean meleeAttackClicked;
+    private boolean closeToAttack;
+    short targetLayer = PhysicsLayer.NPC;
+
+    public PlayerMeleeAttackComponent() {
+        fixtureDef = new FixtureDef();
+    }
+
+    @Override
+    public void create() {
+        if (fixtureDef.shape == null) {
+            logger.trace("{} Setting default bounding box", this);
+            fixtureDef.shape = makeBoundingBox();
+        }
+
+        Body physBody = entity.getComponent(PhysicsComponent.class).getBody();
+        fixture = physBody.createFixture(fixtureDef);
+
+        // if sensor is false, NPC will not be able to collide with player's fixture
+        setSensor(true);
+
+        // event listeners to check if enemy is within range of melee attack
+        entity.getEvents().addListener("collisionStart", this::onEnemyClose);
+
+        playerCombatStats = entity.getComponent(PlayerCombatStatsComponent.class);
+        playerAttackComponent = entity.getComponent(PlayerMeleeAttackComponent.class);
+    }
+
+    /**
+     * Called when collision with any fixture has occurred and sets closeToAttack
+     * to be True. Event system will continuously check collision between objects in world
+     * If enemy is close and player has clicked the attack melee button, damage will be dealt to
+     * the NPC
+     *
+     * @param me is the fixture of current component
+     * @param other is the other fixture which current component is in contact
+     *              with
+     */
+    private void onEnemyClose(Fixture me, Fixture other) {
+
+//        if (playerAttackComponent.getFixture() != me) {
+//            // Not triggered by hitbox, ignore
+//            return;
+//        }
+
+        // By default, should only try detect NPC layers only
+        if (!PhysicsLayer.contains(targetLayer, other.getFilterData().categoryBits)) {
+            // Doesn't match our target layer, ignore - could be obstacle but not NPC
+            return;
+        }
+
+        // Try to detect enemy.
+        Entity target = ((BodyUserData) other.getBody().getUserData()).entity;
```

Comment:
Do we have to check this for null? (may raise a NullReferenceException?)

Reply:
No, I don't think so. Entity target will only be initialized when 2 entities come into contact in game.